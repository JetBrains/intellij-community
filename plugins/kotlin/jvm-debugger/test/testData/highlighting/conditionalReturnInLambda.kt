package conditionalReturnInLambda

// RESUME: 20
fun main() {
  for (x in listOf(37, 42)) {
    //Breakpoint!, conditionalReturn = true, lambdaOrdinal = 2
    oneTwoOne(x, { println("1") }, { if (it == 37) return@oneTwoOne; println("2") })
  }
}

fun oneTwoOne(v: Int, one: (Int) -> Unit, two: (Int) -> Unit) {
  one(v)
  two(v)
  one(v)
}
