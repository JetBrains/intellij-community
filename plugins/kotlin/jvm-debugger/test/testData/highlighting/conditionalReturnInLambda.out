LineBreakpoint created at conditionalReturnInLambda.kt:7 lambdaOrdinal = 2
Run Java
Connected to the target VM
conditionalReturnInLambda.kt:7
Highlight lambda range: return
Disconnected from the target VM

Process finished with exit code 0
